pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -721,11 +721,11 @@ class CreateVmModal extends React.Component {
 
             if (value && value.recommendedResources.ram) {
                 stateDelta.recommendedMemory = value.recommendedResources.ram;
-                const converted = Math.floor(convertToUnit(stateDelta.recommendedMemory, units.B, this.state.memorySizeUnit));
-                if (converted == 0)
+                const converted = convertToUnit(stateDelta.recommendedMemory, units.B, units.GiB);
+                if (converted == 0 || converted % 1 !== 0) // If recommended memory is not a whole number in GiB, set value in MiB
                     this.setState({ memorySizeUnit: units.MiB.name, memorySize: Math.floor(convertToUnit(stateDelta.recommendedMemory, units.B, units.MiB)) });
                 else
-                    this.onValueChanged('memorySize', converted);
```

Comment:
This was actually needed here, because in the onValueChanged function we do check if the host has the available memory before setting it.

Reply:
Fixed